### ISSUE TITLE
Node GUID is not set for VFs by SR-IOV Network Operator

### ISSUE THREAD

**Issue by e0ne** [k8snetworkplumbingwg/sriov-network-operator#267]
The latest SR-IOV Network operator doesn't configure Node GUID for VFs. It means that rdma_cm functionality is broken on SR-IOV.
 

I’m testing ConnectX-6 and get such issue with VFs:

Server-side:

`# ib_write_bw -d mlx5_0 -F -R --report_gbits`

Client-side:

```
# ib_write_bw -d mlx5_0 -F -R --report_gbits 192.168.102.10
Segmentation fault (core dumped)

```
 

dmesg has  following error:

`[ 6587.951447] ib_write_bw[46826]: segfault at 0 ip 00007f3c957ef3b0 sp 00007fff4e973b10 error 4 in librdmacm.so.1.3.37.0[7f3c957e9000+15000]
`
 ibstat output:
```
...
CA 'mlx5_8'
        CA type: MT4126
        Number of ports: 1
        Firmware version: xx.xx.xxxx
        Hardware version: 0
        Node GUID: 0x0000000000000000
        System image GUID: 0x0c42a10300c6cf52
        Port 1:
                State: Down
                Physical state: Disabled
                Rate: 25
                Base lid: 0
                LMC: 0
                SM lid: 0
                Capability mask: 0x00010000
                Port GUID: 0x983854fffeb4df3e
                Link layer: Ethernet
...
```

Everything works without rdma_cm  (-R flag for ib_write_bw). Issue is reproducible both with ConnectX-6 NIC.

 

Environment details:
SR-IOV Network Operator: master
OS: Ubuntu 20.04
Kernel: 5.4.0-100-generic
mlx5_core driver from kernel 
ib_write_bw version: Version: 6.04

**Comment by adrianchiris** [comment]
#268

**Comment by e0ne** [comment]
Fixed by PR #326
